scripts/usb_reset.py

```diff
@@ -212,17 +228,36 @@ def setup_cgroup(domid, pid):
 
     try:
         # unbuffered write to ensure each one is flushed immediately
-        with open(cg_dir + "/tasks", "w", 0) as tasks, \
-                open(cg_dir + "/devices.deny", "w", 0) as deny, \
-                open(cg_dir + "/devices.allow", "w", 0) as allow:
+        # to the kernel's control group filesystem:
+        #
+        # The order of writes is likely not important, but the writes
+        # may have to be a single write() system call for the entire string.
+        #
+        # Using the unbuffered Raw IO mode, we know the write was done
+        # in exactly this way by the write function call itself, not later.
+        #
+        # With such short writes, splitting the writes because of a
+        # too small buffer will not usually happen, but be paranoid
+        # and keep the same code for Python3 too. We've to use binary
+        # mode for Python3 for this as Python3 does now support unbuffered
+        # Raw I/O in Text mode (with implicit unicode str to byte encode()).
+        #
+        # In Python2 str is bytes anyway so in Python2 it was already binary.
+        # Using bytes for "wb" means also no change for use with Python3:
```

Comment:
Ditto, I wrote this comment in a hurry, the comment could be better.
As it was only meant to describe the change for reviewers, it can be dropped for the merge.
Or I could rephrase it, as this last paragraph was only meant for the review.
```suggestion
```